src/main/scala/faces/apps/RandomFaces.scala

```diff
@@ -101,27 +102,45 @@ object RandomFaces extends App {
             centered.camera.principalPoint.y + (2.0 * yTranslationDistribution()) / imageHeight)
         ))
 
+        val imageData =
+          for((postfix, currentRenderer) <- helpers.renderingMethods) yield {
+            if (bg && postfix == "") {
+              require(helpers.loadBgs.nonEmpty, "no Background files with type " + cfg.backgrounds.bgType + " found in " + cfg.backgrounds.bgPath)
+              val rndBG = helpers.loadBgs(rnd.scalaRandom.nextInt(helpers.loadBgs.length))
+              val rndBGimg = PixelImageIO.read[RGBA](rndBG).get.resample(imageWidth, imageHeight)
+              (currentRenderer.renderImage(centered).zip(rndBGimg).map(p => if (p._1.a < 0.5) p._2 else p._1), postfix)
+            }
+            else {
+              (currentRenderer.renderImage(centered), postfix)
+            }
+          }
+
         // render image with or without background
-        val img = if(bg) {
+        /*val img = if(bg) {
```

Comment:
If this code is no longer it should be deleted.

Reply:
yes